meta/store.go

```diff
@@ -1664,6 +1772,33 @@ func (fsm *storeFSM) Apply(l *raft.Log) interface{} {
 	return err
 }
 
+func (fsm *storeFSM) applyRemovePeerCommand(cmd *internal.Command) interface{} {
+	ext, _ := proto.GetExtension(cmd, internal.E_RemovePeerCommand_Command)
+	v := ext.(*internal.RemovePeerCommand)
+
+	id := v.GetID()
+	addr := v.GetAddr()
+
+	// Only do this if you are the leader
+	if fsm.raftState.isLeader() {
+		//Remove that node from the peer
+		fsm.Logger.Printf("removing peer for node id %d, %s", id, addr)
+		if err := fsm.raftState.removePeer(addr); err != nil {
+			fsm.Logger.Printf("error removing peer: %s", err)
+		}
+	}
+
+	// If this is the node being shutdown, stop raft
+	if fsm.id == id {
+		fsm.Logger.Printf("shutting down raft for %s", addr)
+		if err := fsm.raftState.shutdown(); err != nil {
```

Comment:
Instead of just shutting down raft, could transition to a new `droppedState` or similar.  That state would just return errors for all the funcs indicating the node is no longer part of the cluster.